sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JacksonParser.scala

```diff
@@ -81,9 +81,27 @@ private[sql] object JacksonParser {
       case (VALUE_NUMBER_INT | VALUE_NUMBER_FLOAT, FloatType) =>
         parser.getFloatValue
 
+      case (VALUE_STRING, FloatType) =>
+        // Special case handling for NaN and Infinity.
+        val value = parser.getText
+        if (value.equals("NaN") || value.equals("Infinity") || value.equals("-Infinity")) {
```

Comment:
oh, originally I tried scala and it only recognizes these three forms. Just tried Python and it can recognize `nan`. I will make it case insensitive.